File: backend/src/main/java/com/khoinguyen/orderfood/model/Order.java
package com.khoinguyen.orderfood.model;

import com.khoinguyen.orderfood.enums.PaymentMethod;
import com.khoinguyen.orderfood.enums.Status;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "orders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private BigDecimal totalPrice;
    @Column(nullable = false)
    private String phone;
    @Column(nullable = false)
    private String address;
    @Enumerated(EnumType.STRING)
    private Status status = Status.PENDING;

    @Enumerated(EnumType.STRING)
    private PaymentMethod paymentMethod;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<OrderDetail> orderDetails;

    @CreationTimestamp
    private LocalDateTime createdAt;
    @UpdateTimestamp
    private LocalDateTime updatedAt;




}

File: frontend/src/types/Order.ts
import { IProduct } from "./Product";

export type IOrder = {
  id: number;
  user: {
    id: number;
    name: string;
    email: string;
    phone: string;
  };
  totalPrice: number;
  phone: string;
  address: string;
  status: string;
  paymentMethod: string;
  createdAt: string;
  orderDetails: Array<{
    id: number;
    product: IProduct;
    price: number;
    quantity: number;
    message: string;
  }>;
};
